src/reporter/coverage-report.xsl

```diff
@@ -194,56 +191,27 @@
         >
       )
     )
-    (.*)                             <!-- 11: the rest of the string -->
-    $
   </xsl:value-of>
 </xsl:variable>
 
-<!-- TODO: Avoid recursion, because adding @as="node()+" to this xsl:template resurrects
-  xspec/xspec#215 on Saxon 9.9.1.7. -->
-<xsl:template name="test:output-lines">
-  <xsl:context-item use="absent" />
+<xsl:template name="test:output-lines" as="node()+">
+  <xsl:context-item as="element()" use="required" />
 
-  <xsl:param name="line-number" as="xs:integer" required="yes" />
   <xsl:param name="stylesheet-string" as="xs:string" required="yes" />
-  <xsl:param name="node" as="node()" required="yes" />
-  <xsl:param name="number-format" tunnel="yes" as="xs:string" required="yes" />
-  <xsl:param name="module" tunnel="yes" as="xs:string" required="yes" />
+  <xsl:param name="number-format" as="xs:string" required="yes" />
+  <xsl:param name="module" as="xs:string" required="yes" />
+
+  <xsl:variable name="stylesheet-element" as="element()" select="." />
 
-  <!-- $analyzed is a document node created implicitly because its xsl:variable does not have @as -->
-  <xsl:variable name="analyzed">
+  <xsl:variable name="regex-groups" as="map(xs:integer, xs:string)+">

```

Comment:
How about a comment like "Analyze the entire stylesheet string. For each matching substring, create a map that records the kind of match."?